base/shell/explorer/traydeskbtn.cpp

```diff
@@ -82,35 +66,29 @@ HRESULT CTrayShowDesktopButton::DoCreate(HWND hwndParent)
     if (!m_hWnd)
         return E_FAIL;
 
+    // Get desktop icon
     bool bIconRetrievalFailed = ExtractIconExW(L"imageres.dll", -IDI_IMAGERES_DESKTOP, NULL, &m_icon, 1) == UINT_MAX;
     if (bIconRetrievalFailed || !m_icon)
         ExtractIconExW(L"shell32.dll", -IDI_SHELL32_DESKTOP, NULL, &m_icon, 1);
-    
-    m_szIcon.cx = 16;
-    m_szIcon.cy = 16;
-    
-    if (m_icon)
-    {
-        ICONINFO iconInfo;
-        if (GetIconInfo(m_icon, &iconInfo))
-        {
-            BITMAP bmpIcon;
-            if (GetObjectW(iconInfo.hbmMask, sizeof(bmpIcon), &bmpIcon))
-            {
-                m_szIcon.cx = bmpIcon.bmWidth;
-                m_szIcon.cy = bmpIcon.bmHeight;
-                // If this structure defines a black and white icon, this bitmask is formatted
-                // so that the upper half is the icon AND bitmask and the lower half is
-                // the icon XOR bitmask.
-                if (!iconInfo.hbmColor)
-                    m_szIcon.cy /= 2;
-            }
-            DeleteObject(iconInfo.hbmMask);
-            DeleteObject(iconInfo.hbmColor);
-        }
-    }
+
+    // Get appropriate size at which to display desktop icon
+    m_szIcon.cx = GetSystemMetrics(SM_CXSMICON);
+    m_szIcon.cy = GetSystemMetrics(SM_CYSMICON);
+
+    // Prep visual style
     EnsureWindowTheme(TRUE);
 
+    // Get HWND of Taskbar
+    auto parent = GetParent();
```

Comment:
Looking at the ATL: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/atl/atlwin.h;hb=ed7c23fc41ddab7877f9fbc2975d6302a971e1b3#l700
it would appear we are constructing a temporary CWindow object just, in fine, to retrieve handles to windows.
Why not then just use win32 functions directly?


Side note:
In principle, https://learn.microsoft.com/en-us/cpp/atl/reference/cwindow-class?view=msvc-170#getparent seems to indicate there is also a method that returns just the HWND, instead of a CWindow object...